pkg/restore/change_image_repository_action.go

```diff
@@ -0,0 +1,267 @@
+/*
+Copyright 2022 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restore
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/vmware-tanzu/velero/pkg/plugin/framework/common"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+)
+
+const (
+	SPECIFIC = "specific"
+)
+
+// ChangeImageRepositoryAction updates a deployment or Pod's image name
+// if a mapping is found in the plugin's config map.
+type ChangeImageRepositoryAction struct {
+	logger          logrus.FieldLogger
+	configMapClient corev1client.ConfigMapInterface
+}
+
+// NewChangeImageRepositoryAction is the constructor for ChangeImageRepositoryAction.
+func NewChangeImageRepositoryAction(
+	logger logrus.FieldLogger,
+	configMapClient corev1client.ConfigMapInterface,
+) *ChangeImageRepositoryAction {
+	return &ChangeImageRepositoryAction{
+		logger:          logger,
+		configMapClient: configMapClient,
+	}
+}
+
+// AppliesTo returns the resources that ChangeImageRepositoryAction should
+// be run for.
+func (a *ChangeImageRepositoryAction) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"deployment", "statefulsets", "daemonset", "replicaset", "replicationcontroller", "job", "CronJob", "pod"},
```

Comment:
These should be listed in a consistent format. I'm not sure it would affect behavior, but it would be more readable that way:
```
		IncludedResources: []string{"deployments", "statefulsets", "daemonsets", "replicasets", "replicationcontrollers", "jobs", "cronjobs", "pods"},
￼Write
```